src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/NotificationController.java

```diff
@@ -0,0 +1,106 @@
+            Executors.newScheduledThreadPool(1);
+    private HashMap<Integer, ScheduledFuture<?>> handleMap;
+    private NotificationOperations operations;
+    private NotificationPolicy policy;
+
+    public NotificationController(NotificationOperations operations, NotificationPolicy policy) {
+        this.operations = operations;
+        this.policy = policy;
+
+        handleMap = new HashMap<>();
+
+        loadNotificationsFromDatabase();
+
+        EventManager.NOTIFICATION_CREATE.getObservable().subscribe(this::createNotification);
+        EventManager.NOTIFICATION_DELETE.getObservable().subscribe(this::deleteNotification);
+        EventManager.NOTIFICATION_UPDATE.getObservable().subscribe(this::updateNotification);
+    }
+
+    private void loadNotificationsFromDatabase() {
+        List<NotificationRecord> notificationList = operations.getAllNotifications();
+        for (NotificationRecord record : notificationList) {
+            createNotification(new Notification(record.getIdNotification(), record.getName(), record.getDescription(),
+                    record.getSendthreshold(), record.getCheckperiod(), record.getQuery(), policy));
+        }
+    }
+
+    /**
+     * Creates a new Notification from protobuf and adds it to the scheduler
+     *
+     * @param notificationProto the notification to create
+     */
+    public void createNotification(edu.kit.ipd.crowdcontrol.objectservice.proto.Notification notificationProto) {
+        Notification notification = new Notification(notificationProto.getId(), notificationProto.getName(),
+                notificationProto.getDescription(), notificationProto.getSendThreshold(),
+                notificationProto.getCheckPeriod(), notificationProto.getQuery(), policy);
+
+        createNotification(notification);
+    }
+
+    /**
+     * Creates a new Notification and adds it to the scheduler
+     *
+     * @param notification the notification to create
+     */
+    public void createNotification(Notification notification) {
+        final ScheduledFuture<?> notificationHandle =
+                scheduler.scheduleAtFixedRate(notification, 0, notification.getCheckPeriod(), TimeUnit.SECONDS);
+        handleMap.put(notification.getID(), notificationHandle);
+    }
+
+    /**
+     * Deletes a notificationProto
+     *
+     * @param notificationProto the notification to delete
+     */
+    public void deleteNotification(edu.kit.ipd.crowdcontrol.objectservice.proto.Notification notificationProto) {
+        deleteNotification(notificationProto.getId());
+    }
+
+    /**
+     * Deletes a notification
+     *
+     * @param notification the notification to delete
+     */
+    public void deleteNotification(Notification notification) {
+        deleteNotification(notification.getID());
+    }
+
+    private void deleteNotification(int id) {
+        ScheduledFuture<?> notificationHandle = handleMap.get(id);
+        notificationHandle.cancel(true);
```

Comment:
i would set this to false...it doesn't hurt to let this run for one more time and you don't have to be extra careful closing you resources